### PR TITLE
Release mm GitHub docs

### PR DESCRIPTION
**Pull request by mmattar**
### Proposed change(s)

Performs Step 2 and Step 6 of Phase 1 for GitHub reorganization : https://docs.google.com/document/d/1DJ3slGiCHVS1FmROUPUDj-20XimjlLMxP1shy0tc0Vk/edit

Each commit is somewhat self-contained. It may be easier to review this PR, by each commit.

Step 2: Make ML-Agents-Overview the central page for descriptions of each feature. This would move “Overview” content from:
- Training with PPO
- Training with SAC
- Training with Self-Play
- Using Recurrent Neural Networks
- Training with Curriculum Learning
- Training with Imitation Learning
- Training with Environment Parameter Randomization
- Reward Signals
- Training Using Concurrent Unity Instances

Step 6: Move all the hyperparameter descriptions into a single doc (either Training-ML-Agents.md or another). This would end up deleting:
- Training with PPO
- Training with SAC
- Training with Self-Play
- Using Recurrent Neural Networks
- Training with Curriculum Learning
- Training with Imitation Learning
- Training with Environment Parameter Randomization
- Reward Signals
- Training Using Concurrent Unity Instances

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [X] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [X] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by awjuliani** [comment]
Hi @mmattar 

It looks like this PR deletes about half of the documentation. Am I perhaps missing something?

**Comment by mmattar** [comment]
> Hi @mmattar
> 
> It looks like this PR deletes about half of the documentation that is being changed (at least from the lines changed metric and a quick glance at the actual changes). Am I perhaps missing something?

@awjuliani - It does refactor ~9 pages into Training-ML-Agents and ML-Agents-Overview, so those pages are intended to be deleted. It is part of re-structuring. Let me know if it makes things better or worse.

File: README.md [hunk 1]
```diff
@@ -28,17 +28,14 @@ developer communities.
...
- (PPO) and [Soft Actor-Critic](docs/Training-SAC.md)
- (SAC)
+* Two deep reinforcement learning algorithms, Proximal Policy Optimization
+  (PPO) and Soft Actor-Critic (SAC)
 * Support for multiple environment configurations and training scenarios
 * Self-play mechanism for training agents in adversarial scenarios
 * Train memory-enhanced agents using deep reinforcement learning
 * Easily definable Curriculum Learning and Generalization scenarios
-* Built-in support for [Imitation Learning](docs/Training-Imitation-Learning.md) through Behavioral Cloning or Generative Adversarial Imitation Learning
+* Built-in support for Imitation Learning through Behavioral Cloning or Generative Adversarial Imitation Learning
```
Comment (surfnerd) [inline README.md:37]: Do we still have BC, or just GAIL?

Comment (mmattar) [reply README.md:37]: Both.
